src/Umbraco.Web.Common/Media/ImageSharpImageUrlGenerator.cs

```diff
@@ -0,0 +1,100 @@
+        /// Initializes a new instance of the <see cref="ImageSharpImageUrlGenerator" /> class.
+        /// </summary>
+        /// <param name="configuration">The ImageSharp configuration.</param>
+        public ImageSharpImageUrlGenerator(Configuration configuration)
+            : this(configuration.ImageFormats.SelectMany(f => f.FileExtensions).ToArray())
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImageSharpImageUrlGenerator" /> class.
+        /// </summary>
+        /// <param name="supportedImageFileTypes">The supported image file types/extensions.</param>
+        /// <remarks>
+        /// This constructor is only used for testing.
+        /// </remarks>
+        internal ImageSharpImageUrlGenerator(IEnumerable<string> supportedImageFileTypes) => SupportedImageFileTypes = supportedImageFileTypes;
+
+        /// <inheritdoc/>
+        public string? GetImageUrl(ImageUrlGenerationOptions options)
+        {
+            if (options?.ImageUrl == null)
+            {
+                return null;
+            }
+
+            var queryString = new Dictionary<string, string?>();
+
+            if (options.Crop is CropCoordinates crop)
+            {
+                queryString.Add(CropWebProcessor.Coordinates, FormattableString.Invariant($"{crop.Left},{crop.Top},{crop.Right},{crop.Bottom}"));
+            }
+
+            if (options.FocalPoint is FocalPointPosition focalPoint)
+            {
+                queryString.Add(ResizeWebProcessor.Xy, FormattableString.Invariant($"{focalPoint.Left},{focalPoint.Top}"));
+            }
+
+            if (options.ImageCropMode is ImageCropMode imageCropMode)
+            {
+                queryString.Add(ResizeWebProcessor.Mode, imageCropMode.ToString().ToLowerInvariant());
+            }
+
+            if (options.ImageCropAnchor is ImageCropAnchor imageCropAnchor)
+            {
+                queryString.Add(ResizeWebProcessor.Anchor, imageCropAnchor.ToString().ToLowerInvariant());
+            }
+
+            if (options.Width is int width)
+            {
+                queryString.Add(ResizeWebProcessor.Width, width.ToString(CultureInfo.InvariantCulture));
+            }
+
+            if (options.Height is int height)
+            {
+                queryString.Add(ResizeWebProcessor.Height, height.ToString(CultureInfo.InvariantCulture));
+            }
+
+            if (options.Quality is int quality)
+            {
+                queryString.Add(QualityWebProcessor.Quality, quality.ToString(CultureInfo.InvariantCulture));
+            }
+
+            foreach (KeyValuePair<string, StringValues> kvp in QueryHelpers.ParseQuery(options.FurtherOptions))
+            {
+                queryString.Add(kvp.Key, kvp.Value);
+            }
+
+            if (options.CacheBusterValue is string cacheBusterValue && !string.IsNullOrWhiteSpace(cacheBusterValue))
+            {
+                queryString.Add("rnd", cacheBusterValue);
```

Comment:
I think `v` would be a much better (and shorter) query string key, as this isn't a random value we're adding, but a versioning value. Changing this doesn't impact the generation of images: it would only invalidate any images in your browser cache.

And that might actually be for the better, as you should also clear your existing ImageSharp cache as part of the migration, because newly generated images can be different because of the automatic EXIF rotation.

```suggestion
                queryString.Add("v", cacheBusterValue);
```

_This should then also be updated in the corresponding tests._